pkg/frontend/querymiddleware/querysharding.go

```diff
@@ -194,7 +194,19 @@ func newQuery(ctx context.Context, r MetricsQueryRequest, engine *promql.Engine,
 			r.GetQuery(),
 			util.TimeFromMillis(r.GetTime()),
 		)
-
+	case *remoteReadQueryRequest:
+		return engine.NewRangeQuery(
+			ctx,
+			queryable,
+			// We are setting the lookback period to 1 minute and add that amount to
```

Comment:
I would better explain why we can't simply set the lookback delta to 0 (because 0 value means used the default). Instead of 1m I would use 1ns to further signal it's just a trick to workaround it.